app/lib/features/chat_ng/dialogs/message_actions.dart

```diff
@@ -1,19 +1,25 @@
 import 'dart:ui';
 
+import 'package:acter/features/chat_ng/widgets/message_actions_widget.dart';
 import 'package:acter/features/chat_ng/widgets/reactions/reaction_selector.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart'
+    show RoomEventItem;
 import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
 
-// reaction selector action on chat message
-void reactionSelectionAction({
+// message actions on chat message
+void messageActions({
   required BuildContext context,
   required Widget messageWidget,
   required bool isMe,
-  required String roomId,
+  required bool canRedact,
+  required RoomEventItem item,
   required String messageId,
-}) {
-  final RenderBox box = context.findRenderObject() as RenderBox;
-  final Offset position = box.localToGlobal(Offset.zero);
-  final messageSize = box.size;
+  required String roomId,
+}) async {
+  // trigger vibration haptic
+  await HapticFeedback.heavyImpact();
```

Comment:
just a vibration haptic which is often commonly seen on opening message actions .i.e. WhatsApp, Signal etc.